stack/building-with-pip.rst

```diff
@@ -0,0 +1,241 @@
+    ]
+    classifiers = [
+        "Intended Audience :: Developers",
+        "License :: OSI Approved :: BSD License",
+        "Operating System :: OS Independent",
+        "Programming Language :: Python :: 3",
+        "Programming Language :: Python :: 3.10",
+        "Programming Language :: Python :: 3.11",
+    ]
+    keywords = ["lsst"]
+    dependencies = [
+        "lsst-utils",
+        "importlib_resources",
+    ]
+    dynamic = ["version"]
+    requires-python = ">=3.10.0"
+
+The Rubin DM convention is that science pipelines packages that are to be distributed on `PyPI`_ should include the ``lsst-`` prefix in the name.
+This differs from the EUPS naming convention where the ``lsst`` is implicit.
+For example, the ``daf_butler`` EUPS package has a python distribution name of ``lsst-daf-butler``.
+Middleware packages use a dual license for distribution, with the `PyPI`_ package declaring the BSD 3-clause license.
+Most science pipelines packages use the GPLv3 license and for those packages the ``pyproject.toml`` should use:
+
+.. code-block:: toml
+
+    license = {text = "GPLv3+ License"}
+    classifiers = [
+      "License :: OSI Approved :: GNU General Public License v3 or later (GPLv3+)",
+    ]
+
+Every Rubin DM science pipelines package should be owned by the special DM account and that should always be included in the ``authors`` section.
+Additional authors can be included if required.
+
+The ``dependencies`` section can only refer to packages that are available on `PyPI`_ since this is the section that will be read during ``pip install``.
+
+Setuptools Configuration
+------------------------
+
+For ``setuptools`` builds additional configuration is needed so that the python files and data files can be located.
+For example, in ``daf_butler`` there is this configuration:
+
+.. code-block:: toml
+
+    [tool.setuptools.packages.find]
+    where = ["python"]
+
+    [tool.setuptools]
+    zip-safe = true
+    license-files = ["COPYRIGHT", "LICENSE", "bsd_license.txt", "gpl-v3.0.txt"]
+
+    [tool.setuptools.package-data]
+    "lsst.daf.butler" = ["py.typed", "configs/*.yaml", "configs/*/*.yaml"]
+
+    [tool.setuptools.dynamic]
+    version = { attr = "lsst_versions.get_lsst_version" }
+
+This tells ``setuptools`` that the python files are in a ``python/`` directory and what additional non-python files should be included in the distribution.
+
+The license-files section should reflect the specific needs of your package.
+
+When making a `PyPI`_ distribution, the package should work without relying on the EUPS ``$PACKAGE_DIR`` variable being set.
+This means that any supplementary data such as those that would go in a ``config/`` or ``policy/`` directory should instead be included inside the ``python/`` directory and be accessed using the standard package resources APIs.
+These files must then be listed explicitly in the ``package-data`` section of the configuration file.
```

Comment:
Why is `py.typed` in that list?

Reply:
Is that a rhetorical question where you are wanting me to explain why it's there?